builder/comp-builder.nix

```diff
@@ -226,7 +226,13 @@ let
   executableToolDepends =
     (lib.concatMap (c: if c.isHaskell or false
       then builtins.attrValues (c.components.exes or {})
-      else [c]) build-tools) ++
+      else [c]) (build-tools
+        # Detect if hspec-discover or tasty-discover are needed to build the component
+        ++ lib.optional (haskellLib.isTest componentId &&
+          lib.any (x: __elem (x.identifier.name or "") ["hspec" "hspec-discover"]) component.depends) hsPkgs.buildPackages.hspec-discover
```

Comment:
Wait, so this is to support things that incorrectly *depend* on `hspec-discover` rather than putting it in `build-tools`? That seems sad :(

Reply:
Is the issue that cabal puts exes for things in `build-depends` into the environment anyway? What if we had a component option like `unhygienicBuildTools` which would mimic that behaviour, and then we could set it for packages that do things wrong?

Reply:
> Is the issue that cabal puts exes for things in build-depends into the environment anyway?

I have not checked, but I think it might be (a lot of packages seem to have this issue).

> What if we had a component option like unhygienicBuildTools which would mimic that behaviour, and then we could set it for packages that do things wrong?

Would `unhygienicBuildTools` list the tools or the packages that use them?

If we are going to list the packages it might be simpler just to override the build-tools in `modules/configuration-nix.nix`.

Reply:
I was thinking something like:
- `unhygienicBuildTools` component option, set it on packages that *need* this behaviour during their build
- if `unhygienicBuildTools` is set, then at this point in the component builder we dump exes from *all* `build-depends` packages into `executableDepends`

I think making it an option gives us quite a bit of flexibility, e.g. if you want to build all of stackage and you don't want to list every package that has this problem directly, you can just set the global default to true, but for more focussed builds you can either leave it off or turn it on selectively where it's needed.